### PR TITLE
Remove obsolete cudf::strings::replace_nulls

### PR DESCRIPTION
**Pull request by davidwendt**
The `cudf::strings::replace_nulls()` is a public API that was replaced by `cudf::replace_nulls()`. 
The strings one should not be used since the base one handles any column type.

### CODE REVIEW

**Review by codereport — APPROVED** [review]
lgtm

**Review by ttnghia — APPROVED** [review]

**Review by cwharris — APPROVED** [review]

File: cpp/tests/strings/replace_tests.cpp [hunk 1]
```diff
@@ -337,30 +337,6 @@ TEST_F(StringsReplaceTest, ReplaceMulti)
...
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
-  }
-  {
-    auto results = cudf::strings::replace_nulls(strings_view);
-    std::vector<const char*> h_expected{"Héllo", "thesé", "", "ARE THE", "tést strings", ""};
-    cudf::test::strings_column_wrapper expected(h_expected.begin(), h_expected.end());
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
-  }
-}
-
```
Comment (hyperbolic2346) [inline cpp/tests/strings/replace_tests.cpp:363]: Should this test be moved to the base replace nulls or is it covering strings sufficiently now?

Comment (davidwendt) [reply cpp/tests/strings/replace_tests.cpp:363]: The base one seems to have strings well covered.
https://github.com/rapidsai/cudf/blob/d501d2c0b9cc7309f08646a2677c019e6f7906b6/cpp/tests/replace/replace_nulls_tests.cpp#L75-L238

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**